internal/controller/helmrelease_controller.go

```diff
@@ -268,6 +269,7 @@ func (r *HelmReleaseReconciler) reconcileRelease(ctx context.Context, patchHelpe
 
 		msg := fmt.Sprintf("could not get HelmChart object: %s", err.Error())
 		conditions.MarkFalse(obj, meta.ReadyCondition, v2.ArtifactFailedReason, msg)
+		r.Eventf(obj, eventv1.EventSeverityError, v2.ArtifactFailedReason, msg)
```

Comment:
I think this will spam is SC restarts and at upgrades. As all artifacts are removed from status, then re-added.

Reply:
I was referring to Flux upgrades